pisa/core/map.py

```diff
@@ -733,6 +741,14 @@ def plot(self, symm=False, logz=False, vmin=None, vmax=None, backend=None,
 
         X, Y = np.meshgrid(x, y)
         pcmesh = ax.pcolormesh(X, Y, hist.T, **pcolormesh_kw)
+        if not binlabel_format is None:
+            X_mid = np.true_divide(X[1:, 1:] + X[1:, :-1], 2)
```

Comment:
Binning objects provide `midpoints`, that are nicely centered for both linear and log-spaced binnings, so you may want to switch to using those (which will also save a few lines of code here)

Reply:
In this case, x and y have already been linearized in lines 723 to 727, so they are always centered the way this is written right now already without any extra work.

Reply:
actually, `weighted_centers` is what I meant (https://github.com/IceCubeOpenSource/pisa/blob/master/pisa/core/binning.py#L821). this would center your binlabel nicely in the middle of the bin also when bins are logarithmic. The average you're doing right now places it not visually in the middle was my guess

Reply:
They are placed in the middle on linear and logarithmic stages as it is now, because x and y are always made linear before it gets to this point.
`        if to_plot.binning.dims[0].is_log:
            xticks = 2**(np.arange(np.ceil(np.log2(min(x))),
                                   np.floor(np.log2(max(x)))+1))
            x = np.log10(x)
        if to_plot.binning.dims[1].is_log:
            yticks = 2**(np.arange(np.ceil(np.log2(min(y))),
                                   np.floor(np.log2(max(y)))+1))
            y = np.log10(y)`